src/funjson.c

```diff
@@ -208,16 +227,13 @@ FUNCTION(fun_json_query)
     } else {
       if (curr) {
         int len;
-        char *c;
+        char *partialbuf;
         char *jstr = cJSON_PrintUnformatted(curr);
         char *latin1 = utf8_to_latin1(jstr, -1, &len, 0, "json.string");
-        for (c = latin1; *c; c += 1) {
-          if (!isprint(*c) && !isspace(*c)) {
-            *c = '?';
-          }
-        }
-
-        safe_strl(latin1, len, buff, bp);
+            
+        partialbuf = json_unescape_latin1string(latin1,len);
```

Comment:
It looks like every single time you call json_unescape_latin1string you're calling utf8_to_latin1 on manually first to generate the latin to pass to it - I'm not sure why this isn't incorporated into json_unescape_latin1string?, either every time or with a bool arg to specify if it needs to be done first if there are any (potential) cases where you don't need to do this?

Reply:
That is correct, but utf8_to_latin1 takes different arguments depending on where it gets called.